storage/innobase/log/log0recv.cc

```diff
@@ -1635,6 +1635,14 @@ static inline bool check_encryption(page_no_t page_no, space_id_t space_id,
   /* Encryption offset at page 0 is the only way we can identify encryption
   information as of today. Ideally we should have a separate redo type. */
   if (offset == encryption_offset) {
+    /* With Percona Server, Tables created with ENCRYPTION='N' have
+    crypt_data (CRYPT_SCHEME_UNENCRYPTED) in Page 0 with type KEY_MAGIC_PS_V3
+    and size KERYING_ENCRYPTION_INFO_MAX_SIZE rather than INFO_SIZE */
+    if (memcmp(start + 2 + 2, Encryption::KEY_MAGIC_PS_V3,
+               Encryption::MAGIC_SIZE) == 0) {
+      return false;
+    }
```

Comment:
can you plz assert that len == KERYING_ENCRYPTION_INFO_MAX_SIZE / _V2/_V1 ??

Reply:
Also, the commit message messed up the lengths of encryption headers.
KEYRING uses 98 while upstream uses 111. We always ensure our encryption headers do not cross the upstream defined max encryption info size 111.

Reply:
imported PS specific constants and added debug assertion - check https://github.com/percona/percona-xtrabackup/pull/1195/commits/bbb84ee59a1828fe48a23230797bb63a5829b46c